### ISSUE TITLE
create a multi job array

### ISSUE THREAD

**Issue by jtronics** [LaserWeb/LaserWeb4#338]
I'm trying to laser mutiple copies of the same job at once. 
How can I create a job array like in the picture. In moment I do it manual, by copying the different files together.
When lasering multiple objects there should be two modes like in cura for 3d printing): 
- "laser-at-once"
- "one-at-a-time"
Thanks Martin
![job array](https://user-images.githubusercontent.com/638858/26946868-fa9eddea-4c90-11e7-9ca3-1572ff8f94fb.PNG)

**Comment by cprezzi** [comment]
I have a solution for this (kind of).

First, create the folowing macros:
- shiftZeroRight
  - G92X-80 (the distance to shift in x)
  - G0X0Y0
- shiftZeroLeft
  - G92X80
  - G0X0Y0
- shiftZeroUp
  - G92Y-50 (the distance to shift in y)
  - G0X0Y0
- shiftZeroDown
  - G92Y50
  - G0X0Y0

These macros move the origin right/left/up/down by the defined distances and then moves the head to 0/0. This way you can just move the origin and run the job again.

**Comment by tbfleming** [comment]
I wonder if we should add gcode's looping constructs to the sender.

**Comment by jorgerobles** [comment]
I was wondering about svg symbol and redux proxies...

**Comment by cprezzi** [comment]
@tbfleming That could be interesting.

**Comment by pedjas** [comment]
Three year later and seems there is no better solution. I am seconding this request.

It really would be helpful to set cutting rules for an object and then just set pattern to cut number of copies.

**Comment by cprezzi** [comment]
@pedjas You are welcome to implement that and make a pull request.

**Comment by pedjas** [comment]
> 
> 
> @pedjas You are welcome to implement that and make a pull request.

So I have to be programmer not a user to be able to use LaswerWeb?

**Comment by cprezzi** [comment]
No, you can use LaserWeb as it is, like all other users, but if you want to have special features that are not commonly requested, you are welcome to do it yourself. That's open source.

**Comment by pedjas** [comment]
I understand now. You should state in visible place that feature suggestions are not welcome, so we should know.

I never experienced such unpleasant attitude from a developer.

**Comment by cprezzi** [comment]
If this feature would be widely demanded (and a devs had time to develop it), it would have been made. But as you see, you are just the second one who asked for it in more than 3 years. That's not realy a big demanded.

LaserWeb is not actively developed anymore, because most of the devs left the project. I just try to maintain it as good as I can and I am open to let other devs work on it too.

Talking about attitude: It's also not a very pleasant attitude to start your fist post in this issue with a criticism: "Three year later and seems there is no better solution."

**Comment by jamescbury** [comment]
Hi cprezzi - I just came across this thread as I'm looking to do something similar (though I only need to shift my starting point to the right and repeat my gcode).  I am new to laserweb, but really liking the software and would be happy to contribute to it's future development.  I appreciate you keeping this alive!

Specifically for the above, do you think there is a simple way to add the loop via macro?  In my case I want to run my gcode, move 30 mm to the right, reset by XY zero then run the gcode again.  I want to do this for a set amount of loops (i.e, 4-6)

**Comment by cprezzi** [comment]
@jamescbury I did something like this with two macros. First I made sure the job gcode had a move to 0,0 at the end (with "End Gcode" setting). Then I made macro1 that moves x mm in x direction and reset the origin: 

```
G0 X30
G92 X0
```

The second macro moves the origin back to left border and up x mm in y:

```
G0 X-270 Y30
G92 X0 Y0
```

This way you could just hit macro1 or macro2 between each job run.
